### ISSUE TITLE
CData that ends with ]]]> breaks parser

### ISSUE THREAD

**Issue by nicklan** [netvl/xml-rs#98]
Attempting to parse the following valid xml document:

```xml
<root>
  <x><![CDATA[Foo]]></x>
  <y><![CDATA[Foo [Bar]]]></y>
  <z><![CDATA[Quux]]></z>
</root>
```

With the example program (slightly modified to print out the contents of the CData tags) gives the following:

```
+root
    +x
       cdata: Foo
    -x
    +y
       cdata: Foo [Bar]]]></y>
  <z><![CDATA[Quux
Error: 4:25 Unexpected closing tag: z, expected y
```

Seems parser is not reading the ]]]> as a close for the cdata and reading on until the next cdata tag.

**Comment by netvl** [comment]
Thanks! Will look into this ASAP.

**Comment by netvl** [comment]
Fixed in 0.2.0.
